test/cxx_api_tests/RoadRunnerMapTests.cpp

```diff
@@ -0,0 +1,258 @@
+//
+// Created by Ciaran on 09/12/2021.
+//
+
+#include "gtest/gtest.h"
+#include "rrRoadRunnerMap.h"
+#include "rrConfig.h"
+#include "TestModelFactory.h"
+#include <filesystem>
+
+using namespace rr;
+
+class RoadRunnerMapTests : public ::testing::Test {
+public:
+    std::vector<std::string> sbml;
+
+    RoadRunnerMapTests() {
+#ifndef SBMLTestSuiteRoot
+        throw std::logic_error("The SBMLTestSuiteRoot variable is not defined even though it has been "
+                               "set with target_compile_definitions")';
+#endif
+        if (!std::filesystem::exists(SBMLTestSuiteRoot)) {
+            throw std::invalid_argument("Path to sbml test suite does not exist");
+        }
+        Logger::setLevel(Logger::LOG_WARNING);
+
+        // little easy list of sbmls
+        std::vector<std::string> input({
+                                               OpenLinearFlux().str(),
+                                               SimpleFlux().str(),
+                                               Brown2004().str(),
+                                               Venkatraman2010().str()
+                                       });
+        sbml.swap(input);
+    }
+
+    std::string constructSTSXmlFile(int caseId, int level, int version) {
+        std::ostringstream caseOs;
+        std::ostringstream fname;
+
+        caseOs << std::setfill('0') << std::setw(5) << caseId;
+        fname << caseOs.str() << "-sbml-l" << level << "v" << version << ".xml";
+
+        std::filesystem::path root = SBMLTestSuiteRoot;
+        std::filesystem::path caseDir = root / caseOs.str();
+        std::filesystem::path xmlFile = caseDir / fname.str();
+        if (!std::filesystem::exists(xmlFile)) {
+            std::ostringstream os;
+            os << "file not found : " << xmlFile;
+            throw std::logic_error(os.str());
+        }
+        return xmlFile.string();
+    }
+
+    std::vector<std::string> getFirstNModelsFromSTS(int N, int start = 1) {
+        if (start < 1){
+            throw std::invalid_argument("start must be 1 or more");
+        }
+        int end = start + N;
+        int maxSize = 1809;
+        if (end > maxSize){
+            throw std::invalid_argument("start + N must be less than or equal to 1809: " + std::to_string(end));
+        }
+        std::vector<std::string> vec(N);
+        for (int i = start; i < end; i++) {
```

Comment:
There are actually a lot of STS tests without the full gamut of l2v1->l3v2 versions.  There are tests for stuff that was only added in l3; there are tests for stuff that was added in l2v2; there are tests for stuff that was removed in l3, etc. etc.  

In general, I set things up to test the earliest and the latest versions of all the tests (keeping in mind that sometimes there's only one test total).  It looks like here you just want the latest?  If so, you at least should remove the warning lines; they're to be expected.

Reply:
(Actually, I'm removing the warnings myself.  If you indeed only care about grabbing the last one, we're good.)

Reply:
Okay no problem. Since I needed access to the STS in another test suite I've actually moved this content to a little class called `STS` in `<roadrunner-root>/test/sbml-test-suite`. I've only built enough to support my own tests (so will need expanding in future), but in general I think its a good idea to abstract out access to these models to make it easy to quickly grab them. See my later commit.